App/Services/SettingsManager.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Coder.Desktop.App.Services;
+
+/// <summary>
+/// Settings contract exposing properties for app settings.
+/// </summary>
+public interface ISettingsManager
+{
+    /// <summary>
+    /// Returns the value of the StartOnLogin setting. Returns <c>false</c> if the key is not found.
+    /// </summary>
+    bool StartOnLogin { get; set; }
+
+    /// <summary>
+    /// Returns the value of the ConnectOnLaunch setting. Returns <c>false</c> if the key is not found.
+    /// </summary>
+    bool ConnectOnLaunch { get; set; }
+}
```

Comment:
I still think this should just become something like:

```cs
public interface ISettingsManager<T> {
  public Task<T> Read(CancellationToken ct = default); // may return from cache if unchanged, always returns a clone
  public Task Write(T settings, CancellationToken ct = default);
}
```

The setters you currently have use synchronous I/O, which is not something I think you can really work around without having a dedicated method for saving values instead.